### PR TITLE
Move the features (Patterns - Catalog functions, SB Channel, Liveview Metric Store, Analytics Catalog Functions, MQTT Channel, Kafka Streams) to github.

### PR DESCRIPTION
**Pull request by tyogeshp-tibco**

### CODE REVIEW

File: channel/mqtt/pom.xml [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+        <version>1.0</version>
+        <relativePath>../</relativePath>
+    </parent>
+
+	<artifactId>mqtt</artifactId>
+	<name>mqtt</name>
+	<version>1.0.0</version>
+	
+	<properties>
+        <mqtt.version></mqtt.version>
```
Comment (rakulkar-tibco) [inline channel/mqtt/pom.xml:20]: declare and use mqtt version

**Review by rakulkar-tibco — CHANGES_REQUESTED** [review]

File: channel/sb/pom.xml [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+			<artifactId>functions</artifactId>
+			<version>1.4.100.001</version>
+		</dependency>
+		<dependency>
+			<groupId>com.tibco.genxdm.processor.xslt</groupId>
+			<artifactId>processor</artifactId>
+			<version>1.4.100.001</version>
+		</dependency>		
+		<dependency>
+			<groupId>com.tibco.xml</groupId>
```
Comment (rakulkar-tibco) [inline channel/sb/pom.xml:343]: which Mvn repo it is referring to? you need to specify the system path.

File: channel/sb/pom.xml [hunk 3]
```diff
@@ -0,0 +1,517 @@
...
+			<artifactId>cxf.common</artifactId>
+			<version>1.3.500.001</version>
+		</dependency>
+		<dependency>
+			<groupId>com.tibco.xml</groupId>
```
Comment (rakulkar-tibco) [inline channel/sb/pom.xml:348]: same as above. and for all tibco dependencies.

File: channel/sb/pom.xml [hunk 4]
```diff
@@ -0,0 +1,517 @@
...
+			<version>1.4.1</version>
+		</dependency>
+		<dependency>
+			<groupId>org.genxdm</groupId>
+			<artifactId>xslt2.api</artifactId>
+			<version>1.4.100.001</version>
+		</dependency>
+		
+		<dependency>
+			<groupId>com.tibco.as</groupId>
```
Comment (rakulkar-tibco) [inline channel/sb/pom.xml:484]: does it have dependency on AS?

File: channel/sb/pom.xml [hunk 5]
```diff
@@ -0,0 +1,517 @@
...
+			<artifactId>as-common</artifactId>
+			<version>2.4.1</version>
+		</dependency>
+		<dependency>
+			<groupId>com.tibco.as</groupId>
+			<artifactId>as-agent</artifactId>
```
Comment (rakulkar-tibco) [inline channel/sb/pom.xml:490]: same as above

File: channel/sb/pom.xml [hunk 6]
```diff
@@ -0,0 +1,517 @@
...
+			<version>2.4.1</version>
+		</dependency>
+		
+		<!--dependency>
```
Comment (rakulkar-tibco) [inline channel/sb/pom.xml:494]: cleanup this. 
Remove unwanted dependencies in all pom.xmls

File: channel/kafka-streams/src/main/java/com/tibco/cep/driver/kafkastreams/KafkaDriver.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import com.tibco.be.custom.channel.BaseChannel;
+import com.tibco.be.custom.channel.BaseDestination;
+import com.tibco.be.custom.channel.BaseDriver;
+
+/**
+ * Implementation of Channel API's BaseDriver for Kafka.
+ * 
+ * @author moshaikh
+ */
+public class KafkaDriver extends BaseDriver {
```
Comment (rakulkar-tibco) [inline channel/kafka-streams/src/main/java/com/tibco/cep/driver/kafkastreams/KafkaDriver.java:12]: Why KafkaDriver ? Use / copy only required classes from kafka.

File: metric/liveview/pom.xml [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+  <version>1.0.0</version>
+
+  <properties>
+	<liveview.version>1.0.0</liveview.version>
+  </properties>
+
+  <dependencies>
+	<dependency>
+		<groupId>com.tibco.str</groupId>
+		<artifactId>sbclient</artifactId>
```
Comment (rakulkar-tibco) [inline metric/liveview/pom.xml:24]: You need to use System path which will refer to BE installation.

File: metric/liveview/pom.xml [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+		<version>11.0.1</version>
+	</dependency>
+	<dependency>
+		<groupId>com.tibco.str</groupId>
```
Comment (rakulkar-tibco) [inline metric/liveview/pom.xml:28]: same as  above for all tibco dependecies.

File: metric/liveview/pom.xml [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+		<artifactId>lv-client</artifactId>
+		<version>11.0.1</version>
+	</dependency>
+	<!--dependency>
+		<groupId>com.tibco.str</groupId>
```
Comment (rakulkar-tibco) [inline metric/liveview/pom.xml:33]: clean this.

File: catalog/analytics-catalog-fn/pom.xml [hunk 11]
```diff
@@ -0,0 +1,419 @@
...
+            <groupId>com.tibco.businessevents</groupId>
+            <artifactId>cep-common</artifactId>
+        </dependency>
+		<dependency>
+            <groupId>com.tibco.businessevents</groupId>
+            <artifactId>cep-kernel</artifactId>
+        </dependency>
+		
+        <dependency>
+			<groupId>com.tibco</groupId>
```
Comment (rakulkar-tibco) [inline catalog/analytics-catalog-fn/pom.xml:102]: use system which refers to BE installation.

File: catalog/analytics-catalog-fn/pom.xml [hunk 12]
```diff
@@ -0,0 +1,419 @@
...
+			<artifactId>guava</artifactId>
+			<version>23.0</version>
+		</dependency>
+		<dependency>
+			<groupId>joda-time</groupId>
+			<artifactId>joda-time</artifactId>
+			<version>2.9.9</version>
+		</dependency>
+		
+		<!--dependency>
```
Comment (rakulkar-tibco) [inline catalog/analytics-catalog-fn/pom.xml:143]: remove commented dependencies

File: catalog/analytics-catalog-fn/pom.xml [hunk 13]
```diff
@@ -0,0 +1,419 @@
...
+			<artifactId>TIBCrypt</artifactId>
+			<version>5.12.1</version>
+		</dependency>
+		<!--dependency>
+			<groupId>com.tibco.tra</groupId>
+			<artifactId>TIBCOrt</artifactId>
+			<version>5.12.1</version>
+		</dependency-->
+		<dependency>
+			<groupId>com.tibco.tra</groupId>
```
Comment (rakulkar-tibco) [inline catalog/analytics-catalog-fn/pom.xml:187]: For all tibco dependecies use system path referring to BE home,as they are not part of mvn.

File: catalog/analytics-catalog-fn/pom.xml [hunk 14]
```diff
@@ -0,0 +1,419 @@
...
+			<artifactId>io.eclipse</artifactId>
+			<version>5.60.100.001</version>
+		</dependency>
+		<dependency>
+			<groupId>com.tibco.tibcoxml</groupId>
+			<artifactId>net.eclipse</artifactId>
+			<version>5.60.100.001</version>
+		</dependency>
+		<dependency>
+			<groupId>com.tibco.tibcoxml</groupId>
```
Comment (rakulkar-tibco) [inline catalog/analytics-catalog-fn/pom.xml:217]: For all eclipse dependencies reuse dependenies from parent pom.xml

File: catalog/pattern-catalog-fn/pom.xml [hunk 15]
```diff
@@ -0,0 +1,203 @@
...
+			<artifactId>io.eclipse</artifactId>
+			<version>5.60.100.001</version>
+		</dependency>
+		<dependency>
+			<groupId>com.tibco.tibcoxml</groupId>
+			<artifactId>net.eclipse</artifactId>
+			<version>5.60.100.001</version>
+		</dependency>
+		<dependency>
+			<groupId>com.tibco.tibcoxml</groupId>
```
Comment (rakulkar-tibco) [inline catalog/pattern-catalog-fn/pom.xml:162]: Reuse dependencies in parent pom.xml

**Review by rakulkar-tibco — CHANGES_REQUESTED** [review]

File: channel/mqtt/pom.xml [hunk 16]
```diff
@@ -0,0 +1,222 @@
...
+			<version>5.12.1</version>
+		</dependency>
+		<dependency>
+			<groupId>com.tibco</groupId>
+			<artifactId>tibjms</artifactId>
+			<version>10.2.1</version>
+		</dependency>
+		<dependency>
+			<groupId>javax</groupId>
+			<artifactId>jms</artifactId>
```
[the remark this answers is not in the document]
Comment (tyogeshp-tibco) [reply channel/mqtt/pom.xml]: not dependent, removing from pom.

File: channel/kafka-streams/pom.xml [hunk 17]
```diff
@@ -0,0 +1,288 @@
...
+			<version>2.15.0</version>
+		</dependency-->
+		<dependency>
+			<groupId>com.fasterxml.jackson.core</groupId>
+			<artifactId>jackson-annotations</artifactId>
+			<version>2.15.0</version>
+		</dependency>
+		<dependency>
+			<groupId>com.tibco.akd</groupId>
+			<artifactId>tibftl-kafka-avro</artifactId>
```
[the remark this answers is not in the document]
Comment (tyogeshp-tibco) [reply channel/kafka-streams/pom.xml:167]: Included in kafka's classpath.

File: channel/kafka-streams/pom.xml [hunk 18]
```diff
@@ -0,0 +1,288 @@
...
+		</dependency>
+		
+		<dependency>
+			<groupId>io.confluent</groupId>
+			<artifactId>kafka-avro-serializer</artifactId>
+			<version>5.4.3</version>
+		</dependency>
+		<dependency>
+			<groupId>io.confluent</groupId>
+			<artifactId>kafka-schema-registry</artifactId>
```
[the remark this answers is not in the document]
Comment (tyogeshp-tibco) [reply channel/kafka-streams/pom.xml:183]: not required, removing.

File: channel/kafka-streams/pom.xml [hunk 19]
```diff
@@ -0,0 +1,288 @@
...
+			<version>5.4.3</version>
+		</dependency>
+		<dependency>
+			<groupId>io.confluent</groupId>
+			<artifactId>kafka-schema-registry-client</artifactId>
+			<version>5.4.3</version>
+		</dependency>
+		<dependency>
+			<groupId>org.apache.avro</groupId>
+			<artifactId>avro</artifactId>
```
[the remark this answers is not in the document]
Comment (tyogeshp-tibco) [reply channel/kafka-streams/pom.xml:178]: removing avro, mvn clean is showing errors in building the jar.

File: channel/kafka-streams/pom.xml [hunk 20]
```diff
@@ -0,0 +1,288 @@
...
+		</dependency>
+		<dependency>
+			<groupId>org.json</groupId>
+			<artifactId>json</artifactId>
+			<version>20230227</version>
+		</dependency>
+		
+		<dependency>
+			<groupId>io.confluent</groupId>
+			<artifactId>kafka-avro-serializer</artifactId>
```
[the remark this answers is not in the document]
Comment (tyogeshp-tibco) [reply channel/kafka-streams/pom.xml:168]: included in kafka's classpath

**Review by rakulkar-tibco — CHANGES_REQUESTED** [review]

**Review by rakulkar-tibco — COMMENTED** [review]
Also move the examples to be-samples repo

File: channel/sb/pom.xml [hunk 21]
```diff
@@ -0,0 +1,402 @@
...
+    </build>
+
+    <dependencies>
+
+		<dependency>
+			<groupId>com.tibco.str</groupId>
+			<artifactId>sb-java-tools</artifactId>
+			<version>${be.version}</version>
+			<scope>system</scope>
+			<systemPath>${sb.home}/sb-java-tools.jar</systemPath>
```
[the remark this answers is not in the document]
Comment (tyogeshp-tibco) [reply channel/sb/pom.xml]: the property contains the path till ${sb.home}/lib/.

**Pull request merged**